integration-tests/src/test_loop/builder.rs

```diff
@@ -76,6 +86,139 @@ pub(crate) struct TestLoopBuilder {
+        if prev_epoch_protocol_version == protocol_version {
+            return false;
+        }
+
+        // Check the first block height in the epoch separately,
+        // because the block itself is not created yet.
+        // Its relative height is 0.
+        if epoch_manager_adapter.is_next_block_epoch_start(prev_block_hash).unwrap() {
+            return range.contains(&0);
+        }
+
+        // Otherwise we can get start height of the epoch by
+        // the previous hash.
+        let epoch_start_height =
+            epoch_manager_adapter.get_epoch_start_height(&prev_block_hash).unwrap();
+        range.contains(&(height_created as i64 - epoch_start_height as i64))
+    } else if epoch_protocol_version + 1 == protocol_version {
+        // Drop condition for the last epoch with old protocol version.
+        let maybe_upgrade_height = epoch_manager_adapter
+            .get_estimated_protocol_upgrade_block_height(*prev_block_hash)
+            .unwrap();
+
+        // The protocol upgrade height is known if and only if
+        // protocol upgrade happens in the next epoch.
+        let Some(upgrade_height) = maybe_upgrade_height else {
+            return false;
+        };
+        let next_epoch_id =
+            epoch_manager_adapter.get_next_epoch_id_from_prev_block(prev_block_hash).unwrap();
+        let next_epoch_protocol_version =
+            epoch_manager_adapter.get_epoch_protocol_version(&next_epoch_id).unwrap();
+        assert!(epoch_protocol_version < next_epoch_protocol_version);
+        range.contains(&(height_created as i64 - upgrade_height as i64))
+    } else {
+        false
+    }
+}
+
+/// Registers condition to drop certain message received by the peer manager actor.
+fn register_drop_condition(
+    peer_manager_actor: &mut TestLoopPeerManagerActor,
+    chunks_storage: Arc<Mutex<TestLoopChunksStorage>>,
+    epoch_manager_adapter: Arc<dyn EpochManagerAdapter>,
+    condition: &DropCondition,
+) {
+    match condition {
+        DropCondition::ChunksValidatedBy(account_id) => {
+            let account_id = account_id.clone();
+            let drop_chunks_condition = Box::new(
+                move |chunk: ShardChunkHeader,
+                      epoch_manager_adapter: Arc<dyn EpochManagerAdapter>|
+                      -> bool {
+                    let prev_block_hash = chunk.prev_block_hash();
+                    let shard_id = chunk.shard_id();
+                    let height_created = chunk.height_created();
+                    let epoch_id = epoch_manager_adapter
+                        .get_epoch_id_from_prev_block(prev_block_hash)
+                        .unwrap();
+
+                    // Drop chunk if the given account is present in the list
+                    // of its validators.
+                    let chunk_validators = epoch_manager_adapter
+                        .get_chunk_validator_assignments(&epoch_id, shard_id, height_created)
+                        .unwrap();
```

Comment:
This seems a bit random, where do we use this condition? To test chunk validator kickouts? Don't we have a different way for kicking them out now?

Reply:
This is a pre-existing logic, and yeah, this was an old way to test chunk validator kickouts.
In the new version it still seems useful for checking that validator is **not** kicked out, but I don't remember the exact logic.